src/derivatives.jl

```diff
@@ -1,15 +1,15 @@
 #=
 Single-point derivatives of scalar->scalar maps.
 =#
-function finite_difference_derivative(f, x::T, fdtype::Type{T1}=Val{:central},
-    returntype::Type{T2}=eltype(x), f_x::Union{Void,T}=nothing) where {T<:Number,T1,T2}
+function finite_difference_derivative(f, x::T, ::Val{fdtype}=Val{:central}(),
```

Comment:
why instances instead?

Reply:
I switched to instances of Val, because the documentation on value types changed from 0.6 to 0.7:
https://docs.julialang.org/en/stable/manual/types/#"Value-types"-1
https://docs.julialang.org/en/latest/manual/types/#"Value-types"-1

In particular, from the latest docs:

> For consistency across Julia, the call site should always pass a Valinstance rather than using a type, i.e., use `foo(Val(:bar))` rather than `foo(Val{:bar})`.

The stable / current docs reccomend passing around `Val{:central}`, and dispatching using `f(::Type{Val{:central}})`.

Figured I'd update the interface to match the new recommendation, and that instances would eventually become more familiar / expected.

I'll run `@benchmark include("test_files")` after checking out this pr vs current master later to look at performance.
Do you have anything more formal in mind when it comes to regression testing?